chainindex/gc.go

```diff
@@ -0,0 +1,113 @@
+package chainindex
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/lotus/chain/actors/policy"
+)
+
+var (
+	log             = logging.Logger("chainindex")
+	cleanupInterval = time.Duration(8) * time.Hour
+)
+
+func (si *SqliteIndexer) gcLoop() {
+	defer si.wg.Done()
+
+	// Initial cleanup before entering the loop
+	si.cleanupRevertedTipsets(si.ctx)
+	si.gc(si.ctx)
+
+	cleanupTicker := time.NewTicker(cleanupInterval)
+	defer cleanupTicker.Stop()
+
+	for {
```

Comment:
```suggestion
	for ctx.Err() == nil {
```

low probability of being useful, but may as well I think